library/pkwrite.c

```diff
@@ -540,6 +610,38 @@ int mbedtls_pk_write_key_pem( mbedtls_pk_context *key, unsigned char *buf, size_
 
     return( 0 );
 }
+
+int mbedtls_pkcs8_write_key_pem( mbedtls_pk_context *key,
+                                 unsigned char *buf,
+                                 size_t size )
+{
+    int ret;
+    unsigned char output_buf[PRV_PKCS8_DER_MAX_BYTES];
+    const char *begin = PEM_BEGIN_PRIVATE_KEY;
+    const char *end = PEM_END_PRIVATE_KEY;
+    size_t olen = 0;
+    size_t len = 0;
+
+    ret = mbedtls_pkcs8_write_key_der( key, output_buf, sizeof( output_buf ) );
+    if( ret < 0 )
+        return( ret );
+
+    len = (size_t)ret;
+
+    ret = mbedtls_pem_write_buffer( begin,
+                                    end,
+                                    output_buf + sizeof( output_buf ) - len,
+                                    len,
+                                    buf,
+                                    size,
+                                    &olen );
+    if( ret != 0 )
```

Comment:
The algorithm from here until the end of the function can be reduced to `return( ret )`